### DISCUSSION TITLE
i18n for frontend?

### DISCUSSION THREAD

**Discussion by einhirn** [dexidp/dex#2971] [Ideas]
Hi, are there any plans to support i18n for frontend web? Or is it already capable and I didn't find it?

Of course I can change the template to the "main" language here, but I'd like to at least support English and German (in my case) depending on browser language code...

**Comment by nabokihms** [comment]
Hello, @einhirn. We didn't have the feature on our side because Dex's frontend was always only for serving generic purposes with a high level of customization. As you said, you can translate the text via templating either statically, using a template context variable, or dynamically by adding some JS magic to get the browser language. Currently, Dex uses JS sparingly and cannot do i18n properly like modern frontend applications.
